### PR TITLE
planner: add a new sys table to store index advisor results and a new table to store kernel options

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #12303

Problem Summary: planner: add a new sys table to store index advisor results

### What changed and how does it work?

```
CREATE TABLE IF NOT EXISTS mysql.index_advisor_results (
       id bigint primary key not null auto_increment,
       created_at timestamp not null,
       updated_at timestamp not null,
       schema_name varchar(64) not null,
       table_name varchar(64) not null,
       index_name varchar(127) not null,
       index_columns varchar(500) not null COMMENT 'split by ",", e.g. "c1", "c1,c2", "c1,c2,c3,c4,c5"',
       index_details json,        -- est_index_size, reason, DDL to create this index, ...
       top_impacted_queries json, -- improvement, plan before and after this index, ...
       workload_impact json,      -- improvement and more details, ...
       extra json,                -- for the cloud env to save more info like RU, cost_saving, ...
       index idx_create(created_at),
       index idx_update(updated_at),
       unique index idx(schema_name, table_name, index_columns))
```

```
CREATE TABLE IF NOT EXISTS mysql.tidb_kernel_options (
        module varchar(128),
        name varchar(128),
        value varchar(128),
        updated_at datetime,
        status varchar(128),
        description text,
        primary key(module, name))
```

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Review by yudongusa — APPROVED** [review]
LGTM

File: pkg/session/bootstrap.go [hunk 1]
```diff
@@ -722,6 +722,25 @@ const (
...
+       index_name varchar(127) not null,
+       index_columns varchar(500) not null COMMENT 'split by ",", e.g. "c1", "c1,c2", "c1,c2,c3,c4,c5"',
+
+       index_details json,        -- est_index_size, reason, DDL to create this index, ...
+       top_impacted_queries json, -- improvement, plan before and after this index, ...
+       workload_impact json,      -- improvement and more details, ...
+       extra json,                -- for the cloud env to save more info like RU, cost_saving, ...
+       index idx_create(created_at),
+       index idx_update(updated_at),
+       unique index idx(schema_name, table_name, index_columns))`
```
Comment (D3Hunter) [inline pkg/session/bootstrap.go:743]: if a table dropped and add same index using advisor again, will it conflict?

**Review by D3Hunter — APPROVED** [review]

File: pkg/session/bootstrap.go (hunk 1 above)
Comment (qw4990) [reply pkg/session/bootstrap.go:743]: Maybe, but we'll use `insert ... on duplicate ...` to update this table to make sure there is only one record for an unique index.

**Review by 3pointer — APPROVED** [review]

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**